src/alert/alert.tsx

```diff
@@ -19,8 +25,47 @@ export const Alert = ({
     actionLinkIcon,
     sizeType = "default",
     showIcon = false,
+    customIcon,
+    collapsedHeight,
     ...otherProps
 }: AlertProps): JSX.Element => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [showHiddenContent, setShowHiddenContent] = useState<boolean>(false);
+    const [renderShowMore, setRenderShowMore] = useState<boolean>(false);
+    const contentResizeDetector = useResizeDetector<HTMLDivElement>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        setInitialCollapsedState();
+    }, [collapsedHeight]);
```

Comment:
also need to check if the content height > max collapsed height

the content height should be a hook dependency as well